contrib/pyln-testing/pyln/testing/utils.py

```diff
@@ -146,6 +146,27 @@ def get_tx_p2wsh_outnum(bitcoind, tx, amount):
     return None
 
 
+unused_port_lock = threading.Lock()
+unused_port_set = set()
+
+
+def reserve_unused_port():
```

Comment:
One alternative thing we could do is to internalize the `ephemeral_port_reserve` lib (it's just a couple of lines), modify the reservation code not to set `REUSEADDR` itself, so it won't re-reserve a port that is in state `TIME_WAIT`, and then have `lightningd` itself leave the port in state `TIME_WAIT` on shutdown. As long as the time a node is down isn't longer than the `/proc/sys/net/ipv4/tcp_fin_timeout` we should not get the same port when calling `reserve` again.

However this method here is likely just as good, as long as we remember not to use `ephemeral_port_reserve.reserve` directly.